### DISCUSSION TITLE
How to include UIAlertController in a snapshot?

### DISCUSSION THREAD

**Discussion by haemi** [pointfreeco/swift-snapshot-testing#500] [Q&A]
I would like to test the showing of a alert/actionsheet, but obviously it's neither included in the view nor in the viewController, as the UIAlertController is connected to the UIWindow.

Is there a way to include it in a snapshot nevertheless?

**Comment by KacperCzapp** [comment]
Any news on this? It would be very useful to be able to capture it.

**Comment by stephencelis** [comment]
I'm going to convert this to a discussion. While we've been able to capture alerts in screen shots in the past, we've also noticed the ability break over iOS releases :( We're not sure if it's possible these days, but maybe others who've had luck can weigh in there.

**Comment by awaisiosdev** [comment]
I was able to capture alert in the screenshot by matching window layer on iOS 13.7

`assertSnapshot(matching: window!.layer, as: .image)`

**Reply by remlostime** [reply]
@awaisiosdev Could you explain a little bit more about how does it work? I do something like this, but it does not work. :(

```swift
    let vc = ViewController()

    let window = UIWindow(frame: CGRect(x: 0, y: 0, width: 500, height: 800))
    window.rootViewController = vc

    assertSnapshot(matching: window.layer, as: .image, record: true)
```
